### PR TITLE
Map viz: Add pieplot for categorical "overlay"

### LINKED ISSUE #779: Map visualization: donut markers for optional categorical overlay variable
Follows #775 and most likely #776 too.

This is where things get a bit different... (or we need to talk about how to make them more canonical)

We will eventually handle both continuous and categorical overlay variables with mini histograms (or perhaps something simpler) and mini donut markers respectively.

But for now, we will start with just categorical vars.

But what defines the variable constraints for this?  The "map visualization" is primarily driven by the `map-markers` viz of the pass-through computation.  That is where the constraints would normally be defined.

Plan A is to call a second endpoint `geo-pie` (see [ticket](https://github.com/VEuPathDB/EdaDataService/issues/137)) if a categorical overlay variable is chosen, and some other endpoint (e.g. histogram-based) for continuous variables.  The `facetVariable` for these secondary calls would be the geohash variable that was used for the `map-markers` call.

Plan B would be to create a single endpoint that does both - therefore the constraints would be defined in one place.

However, Plan A has advantages of hopefully showing markers relatively quickly from the `map-markers` call and then letting the overlay data come a bit later (it is not quite so time sensitive).

Perhaps it's time to think about formally defining "nested" visualizations like this, so that the constraints can be served from the back end as usual?

### LINKED ISSUE #875: Map: inconsistent behaviour of markers while semantic zooming, and duplicate `map-markers` requests
This ticket assumes that #837 has been completed and merged (fly-to behaviour). _(But note that that work did not introduce this bug. It's just easier to diagnose after a fly-to.)_

Study: SCORE Seasonal Transmission S. haematobium Cluster Randomized Trial

Create a new geo viz, choose communities as geo entity (left hand drop-down).  Let it 'fly to'.

Mouse-over the (33) marker - see that it has quite a large bounding box (grey rectangle).

Double click it.

Sometimes it will just show three markers (incorrectly), other times (or after a small nudge to the map) it will correctly show 16 markers.

This may be connected to the fact that duplicate requests are being made to the back end.  For example, just before the fly-to, the first request is made again.  In fact, after every map movement (or double click on a marker), the previous request is repeated before the new request is made.

### PR DESCRIPTION
**Pull request by bobular**
Depends on https://github.com/VEuPathDB/web-components/pull/310

Resolves #875 
Resolves #779 

NOW WORKS WITH qa/dfalke-b back ends :-) 

Test with **SCORE Seasonal Transmission S. haematobium Cluster Randomized Trial**

### UI issues/questions

1. Is the look/feel of the entity/variable picker consistent enough for now?
2. Ask Outreach if the "output entity" ("Show counts of") selector is really worth having?  It's only needed when an overlay variable hasn't been chosen. It is disabled when not relevant but I haven't provided a tooltip to explain why.
3. We should probably enable the mouse-over zoomed plots, but I think as a separate ticket.
4. Scrolling down the page into the map - not very nice. I think this can be avoided?
5. Layout of entity+variable pickers not quite right. If you select a long variable, e.g. `Schistosoma haematobium, by microscopy` then the two entity choosers squish together.  Removing the `display: 'flex'` from InputVariables' `inputGroup` would solve this, but create other problems!
6. Probably many more questions...

### CODE REVIEW

**Comment by moontrip** [comment]
@bobular I think it works good and I could see donut marker and legend! 👍 
One thing I wonder is whether you would add the function of mouseover pop-up to see larger pie plot (showMouseToolbar prop)

**Comment by bobular** [comment]
> @bobular I think it works good and I could see donut marker and legend! +1 One thing I wonder is whether you would add the function of mouseover pop-up to see larger pie plot (showMouseToolbar prop)
> 
> update: I tried and it worked fine at mapViz, however, I also realized that the mouse control's layout (finger vs magnifying icon at the top right area of the map) and its icons are broken - it shows correctly at web-components though

Yes, see my question 3.

Thanks for looking into this.  Seems like you have volunteered for that task ;-) I'll make a ticket.

**Comment by moontrip** [comment]
@bobular oh, sorry. I didn't read your questions carefully. Yes I can try to do that to make the mouseover event work. 😉

**Review by jtlong3rd — COMMENTED** [review]
Functionality-wise, this is solid!

File: src/lib/core/components/visualizations/VisualizationsContainer.tsx [hunk 1]
```diff
@@ -273,9 +273,10 @@ function FullScreenVisualization(props: Props & { id: string }) {
   const history = useHistory();
   const viz = computation.visualizations.find((v) => v.visualizationId === id);
   const vizType = viz && visualizationTypes[viz.descriptor.type];
-  const overview = visualizationsOverview.find(
-    (v) => v.name === viz?.descriptor.type
+  const overviews = groupBy(visualizationsOverview, (v) =>
```
[the remark this answers is not in the document]
Comment (bobular) [reply src/lib/core/components/visualizations/VisualizationsContainer.tsx]: Thanks - I have done that.
I assume that the result of a `array.find()`, such as `viz`, is also referentially stable?

[the remark this answers is not in the document]
Comment (jtlong3rd) [reply src/lib/core/components/visualizations/VisualizationsContainer.tsx]: > I assume that the result of a `array.find()`, such as `viz` (which is in the dependencies), is also referentially stable?

Yup!

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**